BN#15 feat(view) view specific item

nellykawera opened the pull request:
included productController
included productRoutes
included swagger documentation
[finish #15]

# PR Description
A buyer should be able to see any item from the available items.
# Description of tasks that were expected to be completed
List all the tasks and sub-tasks that were assigned to you Make sure you check every completed task. list them in this style
- [v] Create Given a get request to retrieve the details of a specific item in a sellers collection
# Description of tasks that were expected to be completed and that are not yet completed
- [v] None
# How this may be tested?
Please describe the tests that you ran to verify your functionalities. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.
1. run `npm i` to install all required dependencies
2. run `npm run dev`
3. Swagger/Get
# Screenshots (If appropriate/applicable)
Provide screenshots related to your PR in case it's appropriate and applicable
# Please check this Checklist before you submit your PR:
- [v] My code follows the style guidelines of this project
- [v] I have performed a self-review of my code
- [v] I have commented on my code, particularly in hard-to-understand areas
- [v] My code generates no warnings
- [v] There are no vulnerabilities

dushimeemma commented on src/index.js:
You didn't rebase @nellykawera

dushimeemma commented on src/tests/sum.test.js:
What is the use of this file with this task?

nellykawera replied:
this file is not mine

uwdjany committed:
* feat(profile) create profile feature (#37)

- Profile accessible publicly
- Update profile
- Request validation
[Finishes #36]

* feat(wishlist): create wishlist
- create wishlist controller
- create wishlist routes
- create relationship between product and wishlist models
- validate request body using joi on post request
- validate logged in user
[Finishes #18]

Co-authored-by: Andela Rwanda <andelarwanda@Andelas-MacBook-Pro.local>

EvelyneUmubyeyi commented on src/controllers/productController.js:
```diff
@@ -0,0 +1,51 @@
+    if (!id) {
+      return res.status(404).json({
+        status: 'fail',
+        message: 'Missing id parameter in request.',
```
Could you use translation on all the messages returned in the API response.

Charlesntwari replied:
good work @nellykawera  you putted your effort in this project  but it's also a good practice to not push console's, so i would suggest you to remove them

EvelyneUmubyeyi commented on src/controllers/productController.js:
```diff
@@ -0,0 +1,51 @@
+
+    if (!specificItem) {
+      return res.status(404).json({
+        status: 'fail',
```
You can change status: 'fail' to status: 404

EvelyneUmubyeyi commented on src/controllers/productController.js:
```diff
@@ -0,0 +1,51 @@
+      item: specificItem,
+    });
+  } catch (err) {
+    res.status(400).json({
```
I think status code should be 500 here.

EvelyneUmubyeyi commented on src/controllers/productController.js:
```diff
@@ -0,0 +1,51 @@
+    });
+  }
+};
+
```
I think another endpoint specifically for when a seller wants to retrieve a specific item from their own collection would be needed.

Niyobuzima commented on src/database/seeders/20230326221727-demo-cogito.js:
```diff
@@ -98,7 +98,15 @@ module.exports = {
+=======
+          user_id: 3,
+>>>>>>> 6da50f4 (* feat(profile) create profile feature (#37))
+>>>>>>> e257e9b (* feat(profile) create profile feature (#37))
```
delete this line

Niyobuzima commented on src/database/seeders/20230326221727-demo-cogito.js:
```diff
@@ -98,7 +98,15 @@ module.exports = {
+          user_id: 2,
+=======
+          user_id: 3,
+>>>>>>> 6da50f4 (* feat(profile) create profile feature (#37))
```
and this

Niyobuzima commented on src/database/seeders/20230326221727-demo-cogito.js:
```diff
@@ -98,7 +98,15 @@ module.exports = {
       'wishlists',
       [
         {
+<<<<<<< HEAD
```
delete this line

dushimeemma reviewed (approved):
Good job 🥇
